### PR TITLE
Update: Add widget to dashboard

### LINKED ISSUE #485: New Dashboard Widget Flow

### PR DESCRIPTION
**Pull request by daniellevinson**
Resolves #485 

## Description

When adding a report to a dashboard, instead of transitioning to the report view and having to click "add to dashboard", this will add the new widget and stay in the dashboard view.

## Proposed Changes

- add the new widget and stay in dashboard view
- reuse actions of `report-to-widget` mixin
- switch to controller actions instead of route-actions

## Screenshots

Sep-17-2019 15-37-48

## License

I confirm that this contribution is made under an MIT license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: packages/dashboards/addon/templates/components/navi-dashboard.hbs [hunk 1]
```diff
@@ -100,6 +100,7 @@
         classNames="add-widget primary action"
         reports=(get (get-user) "reports")
         dashboard=dashboard
+        onAddWidgetToDashboard=(action onAddWidgetToDashboard)
```
Comment (jkusa) [inline packages/dashboards/addon/templates/components/navi-dashboard.hbs]: Should this be `onSelectedReport`?

Comment (daniellevinson) [reply packages/dashboards/addon/templates/components/navi-dashboard.hbs]: It's used as the click action handler of the "Add to dashboard" button, so `onAddWidgetToDashboard` is more specific and appropriate imo. The `onchange` of the select only mutates the `selectedReport` property internally.

Comment (jkusa) [reply packages/dashboards/addon/templates/components/navi-dashboard.hbs]: To me `onAddWidgetToDashboard` sounds like the widget was already added to the dashboard, when really the only event that happened was that a report was selected **to be added**.

`onSelect`?

Comment (daniellevinson) [reply packages/dashboards/addon/templates/components/navi-dashboard.hbs]: Oh I see. `onSelect` sounds like an action handler for the select's `onchange`, but I don't have a strong objection here. Let's go with your first suggestion `onSelectedReport`

Comment (daniellevinson) [reply packages/dashboards/addon/templates/components/navi-dashboard.hbs]: updated

**Pull request merged**